src/main/java/io/neonbee/internal/handler/DefaultErrorHandler.java

```diff
@@ -35,31 +38,27 @@
     /**
      * Cached template for rendering the HTML errors.
      */
-    private final String errorTemplate;
+    private String errorTemplate;
 
     /**
-     * Returns a new ErrorHandler with the default {@link #DEFAULT_ERROR_HANDLER_TEMPLATE template}.
-     */
-    public DefaultErrorHandler() {
-        try {
-            this.errorTemplate = readErrorTemplate(DEFAULT_ERROR_HANDLER_TEMPLATE);
-        } catch (IOException e) {
-            throw new RuntimeException("Could not read default error template", e);
-        }
-    }
-
-    /**
-     * Returns a new ErrorHandler with the passed template.
+     * Initializes the ErrorHandler with the passed template.
+     *
+     * @param neonbee the related NeonBee instance
      *
-     * @param errorTemplateName resource path to the template.
-     * @throws IOException in case the template is not found or cannot be read from the current class loader.
+     * @return A Future with the initialized ErrorHandler
      */
-    public DefaultErrorHandler(String errorTemplateName) throws IOException {
-        this.errorTemplate = readErrorTemplate(errorTemplateName);
+    @Override
+    public Future<ErrorHandler> initialize(NeonBee neonbee) {
+        return readErrorTemplate(neonbee).map(errorTemplate -> {
+            this.errorTemplate = errorTemplate;
+            return this;
+        });
     }
 
-    private String readErrorTemplate(String errorTemplateName) throws IOException {
-        return readResourceToBuffer(Objects.requireNonNull(errorTemplateName)).toString();
+    private Future<String> readErrorTemplate(NeonBee neonBee) {
+        String errorTemplatePath = neonBee.getServerConfig().getErrorHandlerTemplate();
+        Path p = errorTemplatePath == null ? DEFAULT_ERROR_HANDLER_TEMPLATE : Path.of(errorTemplatePath);
```

Comment:
Better use a Optional.ofNullable here and make it one statement?

Optional.ofNullable(neonBee.getServerConfig().getErrorHandlerTemplate()).map(Path::of).orElse(DEFAULT_ERROR_TEMPLATE)

Reply:
That is not what Optional is intended for. Using optional here would allocate an object where no object allocation is necessary and to quote Stuart Marks: „It’s generally a bad idea to create an Optional for the specific purpose of chaining methods from it to get a value.“

Reply:
I can see the point of Stuart here. Interesting consideration! Thanks for bringing this up. I think we should add that as a rule to our coding guidelines. I think Optional is used quite broadly in many methods of us.